Library/Homebrew/cli/parser.rb

```diff
@@ -141,6 +143,7 @@ def parse(cmdline_args = ARGV)
         @args_parsed = true
         @args.processed_options = @processed_options
         Homebrew.args = @args
+        Homebrew.args.args_parsed = @args_parsed
```

Comment:
Think it would be good to handle both the above at once and perhaps even return `@args, @args_parsed, @parser` all from this method to be able to make these `attr_reader` instead of `attr_accessor`.

Reply:
I'm not sure I understand this completely. Could you please add in an example code snippet here.

Reply:
```suggestion
```